### PR TITLE
Add unit tests for clear entire costmap and reinitialize global localization BT service nodes

### PR DESCRIPTION
**Pull request by naiveHobo**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #1810 |
| Primary OS tested on | Ubuntu 18.04 |
| Robotic platform tested on | Turtlebot3 Gazebo Simulation |

---

## Description of contribution in a few bullet points

* Added header files for clear entire costmap and reinitialize global localization BT service nodes
* Added unit tests for clear entire costmap and reinitialize global localization BT service nodes

### CODE REVIEW

**Comment by SteveMacenski** [comment]
You have some merge conflicts :-)

File: nav2_behavior_tree/test/plugins/action/test_clear_costmap_service.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      <root main_tree_to_execute = "MainTree" >
+        <BehaviorTree ID="MainTree">
+            <ClearEntireCostmap service_name="clear_entire_costmap"/>
+        </BehaviorTree>
+      </root>)";
+
+  tree_ = std::make_shared<BT::Tree>(factory_->createTreeFromText(xml_txt, config_->blackboard));
+  EXPECT_EQ(config_->blackboard->get<int>("number_recoveries"), 0);
+  EXPECT_EQ(tree_->rootNode()->executeTick(), BT::NodeStatus::SUCCESS);
+  EXPECT_EQ(config_->blackboard->get<int>("number_recoveries"), 1);
```
Comment (SteveMacenski) [inline nav2_behavior_tree/test/plugins/action/test_clear_costmap_service.cpp:107]: I feel like I want more from this, but not sure what else you could do. By this variable incrementing, we know it was called and did something, but I suppose there's no closing the loop.

Comment (naiveHobo) [reply nav2_behavior_tree/test/plugins/action/test_clear_costmap_service.cpp:107]: Ah yes I get what you mean. But this was all I could actually test with the service nodes while being independent from everything else. Again, I definitely agree there should be a proper test to check if the costmap is being cleared correctly, but I'm not sure it belongs here.

Comment (SteveMacenski) [reply nav2_behavior_tree/test/plugins/action/test_clear_costmap_service.cpp:107]: fair enough. We have a recoveries section of nav2_system_tests for the action based ones, but we never test the costmaps, which we should. I'll file a ticket

**Comment by naiveHobo** [comment]
@SteveMacenski is this good to merge?

**Pull request merged**